src/Tools/dotnet-monitor/CollectionRules/Options/Actions/CollectTraceOptions.Validate.cs

```diff
@@ -20,17 +21,32 @@ namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions
     partial record class CollectTraceOptions :
         IValidatableObject
     {
+
+        [MemberNotNullWhen(true, nameof(StoppingEvent))]
```

Comment:
So this attribute allows usage of StoppingEvent without being certain that it's not null correct?

Reply:
This attribute tells the compiler that when `HasStoppingEvent` returns true, the caller can guarantee that `StoppingEvent` is not null.